packages/bot-web-ui/src/pages/dashboard/announcements/announcements.tsx

```diff
@@ -43,20 +44,29 @@ const Announcements = observer(({ is_mobile, is_tablet, handleTabChange }: TAnno
         localStorage?.setItem('bot-announcements', JSON.stringify(updated_local_storage_data));
     };
 
-    const modalButtonAction = (announce_id: string, announcement: TAnnouncement) => () => {
-        setSelectedAnnouncement(announcement);
-        setIsAnnounceDialogOpen(true);
-        setIsOpenAnnounceList(prev => !prev);
-        rudderStackSendAnnouncementClickEvent({ announcement_name: announcement.announcement.main_title });
-
+    const updateLocalStorage = (announce_id: string) => {
         let data: Record<string, boolean> | null = null;
         data = JSON.parse(localStorage.getItem('bot-announcements') ?? '{}');
-
         storeDataInLocalStorage({ ...data, [announce_id]: false });
         const temp_notifications = updateNotifications();
         setReadAnnouncementsMap(temp_notifications);
     };
 
+    const handlePrimaryAction = () => {
+        setActiveTab(1);
+        setFormVisibility(true);
+        updateLocalStorage('UPDATED_QUICK_STRATEGY_MODAL_ANNOUNCE');
+        rudderStackSendAnnouncementClickEvent({ announcement_name: localize('Updated: Quick Strategy Modal') });
+    };
```

Comment:
@rupato-deriv The initial idea of this announcement component was not to include any specific announcement-related functionalities here. Otherwise, if we have like 20/30 announcements then this file would be completely messed up and unmaintainable.

Reply:
done ✅